src/passes/MemoryPacking.cpp

```diff
@@ -721,12 +722,12 @@ void MemoryPacking::createReplacements(Module* module,
         dest =
           builder.makeConstPtr(c->value.getInteger() + bytesWritten, ptrType);
       } else {
-        auto* get = builder.makeLocalGet(-1, Type::i32);
+        auto* get = builder.makeLocalGet(-1, ptrType);
         getVars.push_back(&get->index);
         dest = get;
         if (bytesWritten > 0) {
-          Const* addend = builder.makeConst(int32_t(bytesWritten));
-          dest = builder.makeBinary(AddInt32, dest, addend);
+          Const* addend = builder.makeConst(is64 ? int64_t(bytesWritten) : int32_t(bytesWritten));
```

Comment:
We have a helper for this:
```suggestion
          Const* addend = builder.makeConstPtr(bytesWritten, ptrType);
```